### ISSUE TITLE
SNOW-982536: connection.getQueryStatusThrowIfError() is not throwing an error

### ISSUE THREAD

**Issue by TimShilov** [snowflakedb/snowflake-connector-nodejs#719]
1. What version of NodeJS driver are you using?
1.9.1
   
2. What operating system and processor architecture are you using?
MacOS 14, ARM (Apple M1 Pro)
   
3. What version of NodeJS are you using?
20.10.0

4. What did you do?

I'm trying the new `asyncExec` flag introduced in #672.
* Execute a query with incorrect syntax using `asyncExec: true`
* Get queryId
* Wait for that query to finish
* Run `await connection.getQueryStatusThrowIfError(queryId);`

All the code I'm running I'm talking from the test ([test/integration/testExecuteAsync.js](https://github.com/snowflakedb/snowflake-connector-nodejs/blob/6944e99ed39a44887e87a274f7745b14ec133b97/test/integration/testExecuteAsync.js#L137)). 

5. What did you expect to see?

I expected error to be thrown. Instead I can see that the status is error but it's not thrown.
![Screenshot 2023-12-01 at 13 53 33](https://github.com/snowflakedb/snowflake-connector-nodejs/assets/20239045/dd00c30e-a223-4a67-ab8f-40a4a03969d3)

6. Can you set logging to DEBUG and collect the logs?

```
{"level":"DEBUG","message":"[1:56:10.583 PM]: Contacting SF: /session/v1/login-request?requestId=c8f3c320-ae29-4d79-a7c6-08520d09af32&warehouse=STAGING_WH&databaseName=AGENCY_DEVELOP_TIM&schemaName=PUBLIC, (1/7)"}
{"level":"TRACE","message":"[1:56:10.584 PM]: Create and add to cache new agent https://xe29001.us-central1.gcp.snowflakecomputing.com:443-keepAlive"}
{"level":"TRACE","message":"[1:56:10.584 PM]: CALL POST with timeout 90000: https://xe29001.us-central1.gcp.snowflakecomputing.com/session/v1/login-request?requestId=c8f3c320-ae29-4d79-a7c6-08520d09af32&warehouse=STAGING_WH&databaseName=AGENCY_DEVELOP_TIM&schemaName=PUBLIC"}
{"level":"TRACE","message":"[1:56:10.664 PM]: socket reused = false"}
{"level":"DEBUG","message":"[1:56:10.670 PM]: Reading OCSP cache file. /Users/tim.shilov/Library/Caches/Snowflake/ocsp_response_cache.json"}
{"level":"TRACE","message":"[1:56:10.702 PM]: Returning OCSP status for certificate 06B480EC940D8A0C485E59237E918844 from cache"}
{"level":"TRACE","message":"[1:56:10.704 PM]: Returning OCSP status for certificate 0CF5BD062B5602F47AB8502C23CCF066 from cache"}
{"level":"TRACE","message":"[1:56:10.705 PM]: OCSP validation succeeded for xe29001.us-central1.gcp.snowflakecomputing.com"}
{"level":"DEBUG","message":"[1:56:11.030 PM]: --createStatementPreExec"}
{"level":"DEBUG","message":"[1:56:11.030 PM]: -- binds.length= 0"}
{"level":"DEBUG","message":"[1:56:11.030 PM]: numBinds = 0"}
{"level":"DEBUG","message":"[1:56:11.031 PM]: threshold = 65280"}
{"level":"DEBUG","message":"[1:56:11.031 PM]: RowStatementPreExec"}
{"level":"DEBUG","message":"[1:56:11.031 PM]: context.bindStage=undefined"}
{"level":"TRACE","message":"[1:56:11.031 PM]: Get agent with id: https://xe29001.us-central1.gcp.snowflakecomputing.com:443-keepAlive from cache"}
{"level":"TRACE","message":"[1:56:11.031 PM]: CALL POST with timeout 90000: https://xe29001.us-central1.gcp.snowflakecomputing.com/queries/v1/query-request?requestId=a2cf69f8-2513-480c-9d22-cb33c4793101"}
{"level":"TRACE","message":"[1:56:11.177 PM]: Get agent with id: https://xe29001.us-central1.gcp.snowflakecomputing.com:443-keepAlive from cache"}
{ stillRunning: false }
{"level":"TRACE","message":"[1:56:11.307 PM]: Get agent with id: https://xe29001.us-central1.gcp.snowflakecomputing.com:443-keepAlive from cache"}
isAnError? true
{ status: 'FAILED_WITH_ERROR' }
01b0b2c4-0001-8bfe-0049-210183aea08b

```

**Comment by sfc-gh-dszmolka** [comment]
hi and thank you for raising this issue and providing such a detailed description, appreciated! we're going to take a look

**Comment by sfc-gh-dszmolka** [comment]
possible fix https://github.com/snowflakedb/snowflake-connector-nodejs/pull/722 , under review

**Comment by sfc-gh-ext-simba-jy** [comment]
Hello Team,
The main problem of this function was getQueryStatus function in this function is async and it does not have the await keyword. As a result, the isAnError function cannot notice whether this is an error or not because the status is a Promise. 
I made a PR for this. Please have a look.
https://github.com/snowflakedb/snowflake-connector-nodejs/pull/746

**Comment by sfc-gh-dszmolka** [comment]
#746

**Comment by sfc-gh-dszmolka** [comment]
PR is merged and will be part of the February release, coming in the second half of February

**Comment by sfc-gh-dszmolka** [comment]
fix released with snowflake-sdk 1.10.0, closing this issue

**Comment by Lumengrid** [comment]
we are using the 1.11.0 and types 1.6.24
but the asyncExec parameter is missing
we are using nodejs \ typescript

**Comment by sfc-gh-dszmolka** [comment]
hi - the external types you're using (v1.6.24) is not maintained by Snowflake
however, sadly it is indeed missing in the internal types (too); but it will be fixed with #851
